tests/codegen/issues/issue-112169.rs

```diff
@@ -0,0 +1,28 @@
+//@ compile-flags: -O
+#![crate_type = "lib"]
+
+// Test that simple iterator-based loops of length > 101 are fully optimized away.
+// See <https://github.com/rust-lang/rust/issues/112169>
+
+// CHECK-LABEL: define{{.*}}i32 @issue_112169(){{.*}}{
+#[no_mangle]
+pub fn issue_112169() -> i32 {
+    // CHECK-NOT: icmp
+    // CHECK-NOT: add
+    // CHECK-NOT: br
+    // CHECK: ret i32 102
+    // CHECK-NOT: icmp
+    // CHECK-NOT: add
+    // CHECK-NOT: br
```

Comment:
Instead of checking for the absence of various conditionals, we can just check if the `ret` immediately follows the label
```suggestion
    // CHECK-NEXT: ret i32 102
```

Reply:
Using
```
// CHECK-LABEL: @issue_112169
#[no_mangle]
pub fn issue_112169() -> i32 {
    // CHECK-NEXT: {{.*}}:
    // CHECK-NEXT: ret i32 102
    let mut s = 0;
```
works for me. Check for the function definition, any block label, then a ret

Reply:
There is always a label there - it looks like the printing machinery just omits it in functions with only one block (if you use Godbolt's 'show opt pipeline' which shows the raw output the label is present). Since functions will *always* begin with a label, it's safe to put that there

Reply:
The label for the entry block is omitted if it's unnamed. But I think this should not happen with `--emit=llvm-ir`, so this is probably fine.